flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/table/module/StateModule.java

```diff
@@ -0,0 +1,123 @@
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.table.functions.FunctionKind.TABLE;
+
+/** Module of state in Flink. */
+@Experimental
+public class StateModule implements Module {
+
+    public static final String IDENTIFIER = "state";
+
+    public static final BuiltInFunctionDefinition SAVEPOINT_METADATA =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("savepoint_metadata")
+                    .kind(TABLE)
+                    .runtimeClass(SavepointMetadataTableFunction.class.getName())
+                    .outputTypeStrategy(
+                            TypeStrategies.explicit(
+                                    DataTypes.ROW(
+                                            DataTypes.FIELD(
+                                                    "checkpoint-id", DataTypes.BIGINT().notNull()),
+                                            DataTypes.FIELD("operator-name", DataTypes.STRING()),
+                                            DataTypes.FIELD("operator-uid", DataTypes.STRING()),
+                                            DataTypes.FIELD(
+                                                    "operator-uid-hash",
+                                                    DataTypes.STRING().notNull()),
+                                            DataTypes.FIELD(
+                                                    "operator-parallelism",
+                                                    DataTypes.INT().notNull()),
+                                            DataTypes.FIELD(
+                                                    "operator-max-parallelism",
+                                                    DataTypes.INT().notNull()),
+                                            DataTypes.FIELD(
+                                                    "operator-subtask-state-count",
+                                                    DataTypes.INT().notNull()),
+                                            DataTypes.FIELD(
+                                                    "operator-coordinator-state-size-in-bytes",
+                                                    DataTypes.BIGINT().notNull()),
+                                            DataTypes.FIELD(
+                                                    "operator-total-size-in-bytes",
+                                                    DataTypes.BIGINT().notNull()))))
+                    .build();
+
+    public static final StateModule INSTANCE = new StateModule();
+
+    private final Map<String, BuiltInFunctionDefinition> normalizedFunctions;
+    private final Set<String> functionNamesWithInternal;
+    private final Set<String> functionNamesWithoutInternal;
+
+    public StateModule() {
```

Comment:
We could make constructor `private` since we are using singleton

Reply:
Fixed.